src/oneD/MultiNewton.cpp

```diff
@@ -232,6 +232,19 @@ int MultiNewton::dampStep(const double* x0, const double* step0,
 
     // damping coefficient starts at 1.0
     double damp = 1.0;
+
+
+    // Compute the residual norm (using temporary variables)
+    double temp_x[m_n];  // Temporary array for x0
+    double temp_step[m_n]; // Temporary array for step0
```

Comment:
I had to use the temporary ones because of a complaint about const arguments to the ssnorm(). I thought that the norm operation shouldn't have any side effects, but I resisted changing the arguments to const.

Reply:
You may just not want to use the `ssnorm` function. It ends up making another call to `eval`, when at least in the first instance, you already know the value of the residual (I think). And in the second case, I think there is a later call to `eval` with the new value of `x` that would become redundant.

Before going any further on this though, I think it's worth taking Ingmar's suggestion and pulling these algorithmic changes out into a separate branch / PR.